pubpub/pubpub
# Move DOI prefixes and Crossref credentials to database

This PR moves hard-coded DOI prefixes from [our code](https://github.com/pubpub/pubpub/blob/master/utils/crossref/communities.js)—and Crossref credentials from our environment variables—into a new database model called `DepositTarget`.

**Test Plan**

PubPub prefix
- Create a DOI for a Pub on duqduq. It should have the prefix `10.21428`
- Deposit the Pub. It should work!

Managed Prefix
- Log in to a community that has a custom DOI prefix, like Projections
- Go to a Pub settings dashboard page, e.g. `/dash/pub/2w805oes/settings` (Projections)
- The DOI prefix in the **DOI** section should match that community's custom prefix, e.g. `10.1162` (Projections), and the Pub should be depositable

Unmanaged Prefix
- Visit my test community `mcd` (this community has a custom prefix but no DepositTarget credentials)
- Create a Pub and publish a release
- Go to the Pub settings dashboard Page
- The Pub should not be depositable with the warning `Ths Pub cannot be deposited to Crossref because this Community's DOI prefix is not managed by PubPub.`
